airflow/models.py

```diff
@@ -3678,7 +3757,7 @@ def refresh_from_db(self, session=None):
             DR.dag_id == self.dag_id,
             DR.execution_date == self.execution_date,
             DR.run_id == self.run_id
-        ).one()
+        ).first()
```

Comment:
This function is only called in one place, where it is expected to always
return something (it actually tries to retrieve a record inserted a few
lines before so I actually don't see any situation in which it returns
either None or throws an exception

On Wednesday, September 7, 2016, bolkedebruin <notifications@github.com>
wrote:

> In airflow/models.py
> <https://github.com/apache/incubator-airflow/pull/1730#discussion_r77947936>
> :
>
> > @@ -3678,7 +3757,7 @@ def refresh_from_db(self, session=None):
> >              DR.dag_id == self.dag_id,
> >              DR.execution_date == self.execution_date,
> >              DR.run_id == self.run_id
> > -        ).one()
> > +        ).first()
>
> I know. Why did you change it? Did you check all other code whether it is
> expecting a Exception or a None?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-airflow/pull/1730/files/a03fe0478cf7627255782696f083de4deeb8640e#r77947936>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGYMpqSZEYFiwDcsS44FhjbUXC4wHdq_ks5qn52DgaJpZM4JjjUq>
> .
>


-- 
Thank you,

Norman Mu